src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/HtmlSmartIndentFormatOnTypeProvider.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Formatting
+{
+    internal class HtmlSmartIndentFormatOnTypeProvider : RazorFormatOnTypeProvider
+    {
+        public override string TriggerCharacter => "\n";
```

Comment:
Does this work on files with CRLF?

Reply:
Yes only because the last character is `\n` 😄